p4-16/spec/P4-16-spec.mdk

```diff
@@ -2208,9 +2208,12 @@ structField
 ~ End P4Grammar
 
 where each `typeRef` is restricted to a bit-string type (fixed or
-variable) or an integer type. This declaration introduces a new
-identifier in the current scope; the type can be referred to using
-this identifier. A header is similar to a `struct` in C,
+variable), an integer type, a boolean type, or a struct that itself contains other 
+struct fields, nested arbitrarily, as long as all of the "leaf" types are
+are `bit<W>`, `int<W>`, a Serializable Enum, or a `bool`.  If a `bool` is used
+inside a P4 header, all implementations encode the `bool` as a one bit long field. 
+This declaration introduces a new identifier in the current scope; the 
+type can be referred to using this identifier. A header is similar to a `struct` in C,
```

Comment:
Nate was asking for emphasizing that nested headers are not supported - if this isn't in the text already.

Reply:
The text does not exist in the spec.  I will add it.